lib/pluginmanager/unpack.rb

```diff
@@ -0,0 +1,51 @@
+require "logstash/util/compress"
+require "fileutils"
+
+class LogStash::PluginManager::Unpack < LogStash::PluginManager::Command
+
+  option "--tgz", :flag, "unpack a packaged tar.gz file", :default => !LogStash::Environment.windows?
+  option "--zip", :flag, "unpack a packaged  zip file", :default => LogStash::Environment.windows?
+  option "--force", :flag, "Override any previously generated cache", :default => false
+
+  parameter "file", "the package file name", :attribute_name => :package_file, :required => true
+
+  def execute
+    puts("Unpacking #{package_file} to be used as a plugins local cache")
+
+    cleanup(LogStash::Environment::CACHE_PATH) if force?
+    validate_cache_location
+    unpack(package_file)
+
+    puts("Unpacked at #{LogStash::Environment::CACHE_PATH}")
+    puts("If you want to use the generated cache to install/update plugins you should call bin/plugin with the --local flag")
+  end
+
+  private
+
+  def validate_cache_location
+    location = LogStash::Environment::CACHE_PATH
+    if File.exist?(location)
+      puts("Directory #{location} is going to be override, do you want to continue? (Y/N)")
+      override = ( "y" == STDIN.gets.strip.downcase ? true : false)
+      if override
+          cleanup(location)
+      else
+        puts("Unpack cancelled, a previously generated cache exist at location: #{location}, move it to a save place and run the command again")
+        exit
+      end
+    end
+  end
+
+  def unpack(file)
+    if zip?
+      LogStash::Util::Zip.extract(file, LogStash::Environment::CACHE_PATH)
+    else
+      LogStash::Util::Tar.extract(file, LogStash::Environment::CACHE_PATH)
+    end
+  end
+
+  def cleanup(location)
+    FileUtils.rm_rf(location)
+  end
```

Comment:
I would just use `FileUtils.rm_rf` directly, since in the `package.rb` we also use `FileUtils`, I am not seeing the value of this method since it take only 1 argument and pass it to `#FileUtils.rm_rf` ?